### PR TITLE
feat(tsx): use swc

### PR DESCRIPTION
**Pull request by AugustinMauroy**
* reduce dependency size by 20KO
* it's supposed to perform better

https://swc.rs/docs/benchmarks

### CODE REVIEW

File: tsx.mjs [hunk 1]
```diff
@@ -38,37 +28,42 @@ export async function load(url, ctx, nextLoad) {
...
+          development: true,
+          useBuiltins: true,
+        },
       }
+    }
+  };
 
-      return {};
-    });
+  //console.log('config', config);
```
Comment (AugustinMauroy) [inline tsx.mjs:57]: self review remove that

**Comment by JakobJingleheimer** [comment]
The benchmarks on that link appear to be broken: there's only 1 bar or line in the charts.

**Comment by AugustinMauroy** [comment]
> The benchmarks on that link appear to be broken: there's only 1 bar or line in the charts.

did you opened an issue on their repo ?

**Comment by JakobJingleheimer** [comment]
No, not yet. Wanted to check with you because you saw it working before (so checking with you that what you saw before is not what you see now).

**Comment by AugustinMauroy** [comment]
@JakobJingleheimer should I also syncify tsx on this pr ????

**Comment by JakobJingleheimer** [comment]
> > The benchmarks on that link appear to be broken: there's only 1 bar or line in the charts.
> 
> did you opened an issue on their repo ?

This was apparently a choice: https://github.com/swc-project/swc/issues/8882#issuecomment-2069265588

**Comment by JakobJingleheimer** [comment]
I would not combine those changes for a couple reasons:

1. Syncification has not landed in node, so this hook would be broken when used with another hook that is async (because this hook would fail to await the other).
2. They can't be backed out independently, and they're not related.

I already have a PR to syncify the hooks btw: https://github.com/JakobJingleheimer/nodejs-loaders/pull/16. Syncifying introduces a non-trivial performance degradation (which I raised to node), which needs to be explored.

File: tsx.mjs [hunk 2]
```diff
@@ -38,37 +28,40 @@ export async function load(url, ctx, nextLoad) {
...
+        },
       }
+    }
+  };
 
-      return {};
-    });
-
-  if (warnings?.length) console.warn(...warnings);
+  const { code: source } = await transform(rawSource, config);
```
[the remark this answers is not in the document]
Comment (AugustinMauroy) [reply tsx.mjs:57]: from swc docs : "A swcrc value passed in the programmatic options will override one set within a configuration file."

File: tsx.mjs [hunk 3]
```diff
@@ -38,37 +28,40 @@ export async function load(url, ctx, nextLoad) {
...
+          useBuiltins: true,
+        },
       }
+    }
+  };
 
-      return {};
-    });
-
-  if (warnings?.length) console.warn(...warnings);
```
[the remark this answers is not in the document]
Comment (AugustinMauroy) [reply tsx.mjs:55]: Capture d’écran 2024-08-08 à 12 28 34

SWC already do it for us

File: tsx.spec.mjs [hunk 4]
```diff
@@ -59,24 +59,26 @@ describe('JSX & TypeScript loader', { concurrency: true }, () => {
 			});
 		});
 
-		// This verifies that esbuild.config.mjs is being loaded correctly because the one in this repo
-		// disables minification, but the loader's default config enables it.
```
[the remark this answers is not in the document]
Comment (AugustinMauroy) [reply tsx.spec.mjs:63]: Done ç

File: tsx.mjs (hunk 2 above)
[the remark this answers is not in the document]
Comment (AugustinMauroy) [reply tsx.mjs:61]: FIY, it's now documented on our code

**Comment by AugustinMauroy** [comment]
The decision in the project is not to support/create a loader if one is already official.
https://github.com/swc-project/swc-node/tree/master/packages/register#swc-noderegister

**Pull request closed without merging**